trace/config.go

```diff
@@ -135,71 +185,56 @@ func (o attributeSpanOption) apply(c *SpanConfig) {
 // If multiple of these options are passed the attributes of each successive
 // option will extend the attributes instead of overwriting. There is no
 // guarantee of uniqueness in the resulting attributes.
-func WithAttributes(attributes ...attribute.KeyValue) LifeCycleOption {
-	return attributeSpanOption(attributes)
+//nolint:golint
+func WithAttributes(attributes ...attribute.KeyValue) attributeOption {
```

Comment:
This is returning an unexported type and is not following our configuration style guide.

Reply:
Doing so would mean that we would have another Interface type that is only used as the return of these two functions.  We can achieve the same thing by return the concrete type here, all while not adding any extra code to maintain.

Reply:
I would prefer adding exported types that conform with our style guidelines rather than return an unexported type.

Reply:
Using exported types will help discoverability when users are reading documentation. I'm up for discussing more in the SIG meeting tomorrow if you want.